File: release/SmartLibrarySystem_Delivery_20260320/front_library/services/api/reviewService.ts
import apiClient from "@/lib/axios";
import { ApiReviewCreateDto, ApiReviewDto, PageResponse } from "@/types/api";

export const reviewService = {
  /**
   * 获取图书公开评论（仅 APPROVED）
   * GET /api/books/{bookId}/reviews
   */
  getBookReviews: async (
    bookId: number,
    page = 0,
    size = 20,
  ): Promise<PageResponse<ApiReviewDto>> => {
    const { data } = await apiClient.get<PageResponse<ApiReviewDto>>(
      `/books/${bookId}/reviews`,
      { params: { page, size, sortBy: "createTime", direction: "DESC" } },
    );

    return data;
  },

  /**
   * 提交评论
   * POST /api/reviews
   */
  createReview: async (payload: ApiReviewCreateDto): Promise<ApiReviewDto> => {
    const { data } = await apiClient.post<ApiReviewDto>("/reviews", payload);

    return data;
  },

  /**
   * 我的评论
   * GET /api/reviews/me
   */
  getMyReviews: async (
    page = 0,
    size = 20,
  ): Promise<PageResponse<ApiReviewDto>> => {
    const { data } = await apiClient.get<PageResponse<ApiReviewDto>>(
      "/reviews/me",
      { params: { page, size } },
    );

    return data;
  },

  /**
   * 更新评论
   * PUT /api/reviews/{id}
   */
  updateReview: async (
    reviewId: number,
    payload: ApiReviewCreateDto,
  ): Promise<ApiReviewDto> => {
    const { data } = await apiClient.put<ApiReviewDto>(
      `/reviews/${reviewId}`,
      payload,
    );

    return data;
  },

  /**
   * 删除评论
   * DELETE /api/reviews/{id}
   */
  deleteReview: async (reviewId: number): Promise<void> => {
    await apiClient.delete(`/reviews/${reviewId}`);
  },

  /**
   * 获取待审核评论（管理员）
   * GET /api/admin/reviews/pending
   */
  getPendingReviews: async (
    page = 0,
    size = 10,
  ): Promise<PageResponse<ApiReviewDto>> => {
    const { data } = await apiClient.get<PageResponse<ApiReviewDto>>(
      "/admin/reviews/pending",
      { params: { page, size } },
    );

    return data;
  },

  /**
   * 获取评论审核列表（管理员）
   * GET /api/admin/reviews
   */
  getAdminReviews: async (
    page = 0,
    size = 10,
    options?: {
      status?: "PENDING" | "APPROVED" | "REJECTED";
      keyword?: string;
    },
  ): Promise<PageResponse<ApiReviewDto>> => {
    const params: Record<string, unknown> = { page, size };

    if (options?.status) {
      params.status = options.status;
    }
    if (options?.keyword?.trim()) {
      params.keyword = options.keyword.trim();
    }

    const { data } = await apiClient.get<PageResponse<ApiReviewDto>>(
      "/admin/reviews",
      { params },
    );

    return data;
  },

  /**
   * 审核评论（管理员）
   * PUT /api/admin/reviews/{id}/audit?approved=true|false
   */
  auditReview: async (
    reviewId: number,
    approved: boolean,
  ): Promise<ApiReviewDto> => {
    const { data } = await apiClient.put<ApiReviewDto>(
      `/admin/reviews/${reviewId}/audit`,
      null,
      { params: { approved } },
    );

    return data;
  },
};

File: release/SmartLibrarySystem_Delivery_20260320_230043/backend-library/src/main/java/com/example/library/service/ReviewService.java
package com.example.library.service;

import com.example.library.dto.ReviewDto;
import com.example.library.dto.ReviewResponseDto;
import com.example.library.entity.BookReview;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * Review service.
 */
public interface ReviewService {
    /**
     * Creates a review for a book.
     */
    ReviewResponseDto createReview(Integer userId, ReviewDto reviewDto);

    /**
     * Returns approved reviews for a book.
     */
    Page<ReviewResponseDto> getReviewsByBookId(Integer bookId, Pageable pageable);

    /**
     * Updates review moderation status.
     */
    void updateReviewStatus(Long reviewId, BookReview.ReviewStatus status);

    /**
     * Returns pending reviews for admin audit.
     */
    Page<ReviewResponseDto> getPendingReviews(Pageable pageable);

    /**
     * Returns admin review list with optional status and keyword filters.
     */
    Page<ReviewResponseDto> getAdminReviews(BookReview.ReviewStatus status, String keyword, Pageable pageable);

    /**
     * Approves or rejects a review.
     */
    ReviewResponseDto auditReview(Integer reviewId, boolean approved);

    /**
     * Returns reviews submitted by a specific user.
     */
    Page<ReviewResponseDto> getReviewsByUser(Integer userId, Pageable pageable);

    /**
     * Updates a review's content.
     */
    ReviewResponseDto updateReview(Integer reviewId, ReviewDto reviewDto);

    /**
     * Deletes a review.
     */
    void deleteReview(Integer reviewId);
}
